subsys/bluetooth/controller/hal/nrf5/include/hal_radio_nrf51.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2018 Ioannis Glaropoulos
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef _HAL_RADIO_NRF51_H_
+#define _HAL_RADIO_NRF51_H_
+
+#include <soc.h>
```

Comment:
Do not include inside header files (hence, dont need guards)

Reply:
These header files do not only declare MACROS but also define functions, i.e. need type definitions, like u8_t. So in theory they would need inclusions.
Of course, since these files are not to be exposed we could include them in the right place, so as not to need any type definitions to be included....
But I just thought that in this way we make these headers a bit more self-sustained.